### PR TITLE
Addition of surface variable for the Cool skin operator

### LINKED ISSUE #105: Make use of the CoolSkin UFO
Add the necessary surface fields to make use of @HamidehGMAO 's cool skin simulator.

### PR DESCRIPTION
**Pull request by guillaumevernieres**
This branch requires the ufo feature/marine_cs_coolskin_gv to build and run (not tested with intel). We will have to wait until @HamidehGMAO 's PR is reviewed before we can merge the soca counterpart.
Feature added:
- Use of @HamidehGMAO 's cool skin ufo
- 5 new surface variable
- What needs more work:
        a - Covariance model for new variable is the identity
        b - No realistic background error
        c - read_file method in ocnsfc is a hack, this will be fixed in the next pr
        d - ...  
Considering that the new variables are not used to update the model state a & b can be ignored for the time being.   

closes #105

### CODE REVIEW

**Comment by travissluka** [comment]
The surface fluxes have no horizontal variation. When I was setting up the 4deg grid I just zeroed out all forcings, because at the time I figured 1) we didn't need them for unit testing and 2) I was lazy and didn't want to regrid the existing climatological forcing file. In order to have proper ctests with the cool skin obs operator, a horizontally varying flux file should to be added, I'll do this today/tomorrow in a separate branch.

Also, nothing using cool skin works right now because it is seg faulting in ufo.

**Comment by guillaumevernieres** [comment]
@travissluka , by design they don't have any horiz variation because I'm setting them to a bogus # while reading. I just want to merge the plumbing part for now.

Are you using the ufo branch I specified above? All test that should pass are passing for me ...

**Comment by guillaumevernieres** [comment]
Let me retest with @HamidehGMAO 's ufo branch and the last merge in soca develop. I'll send a note when it's ready to be reviewed again.

**Comment by guillaumevernieres** [comment]
Green light!
But switch to @HamidehGMAO 's feature/marine_cs_coolskin_HE_cu ufo branch. All tests that should be ok are ok (container/gcc 7.4)

**Comment by guillaumevernieres** [comment]
Update your bundles next time you are reviewing this PR, @HamidehGMAO 's branch was merged into ufo develop.

File: test/testoutput/3dvar_soca.test [hunk 1]
```diff
@@ -1,25 +1,32 @@
...
-Test     : CostJo   : Nonlinear Jo(SeaSurfaceSalinity) = 67.154450, nobs = 51, Jo/n = 1.316754, err = 1.000000
-Test     : CostJo   : Nonlinear Jo(ADT) = 189.483262, nobs = 99, Jo/n = 1.913972, err = 0.100000
-Test     : CostJo   : Nonlinear Jo(InsituTemperature) = 242.454975, nobs = 206, Jo/n = 1.176966, err = 0.900203
-Test     : CostJo   : Nonlinear Jo(InsituSalinity) = 68.312997, nobs = 218, Jo/n = 0.313362, err = 0.608197
-Test     : CostJo   : Nonlinear Jo(SeaIceFraction) = 145.772125, nobs = 85, Jo/n = 1.714966, err = 0.100000
-Test     : CostFunction: Nonlinear J = 17040.853159
+Test     :      sw   min=  -57.179584   max=  -57.039606   mean=  -86.521766
+Test     :     lhf   min=  102.803483   max=  103.258957   mean=  156.072304
+Test     :     shf   min=    7.698902   max=    7.701447   mean=   11.667577
+Test     :      lw   min=  599.741487   max=  600.196180   mean=  909.162621
```
Comment (travissluka) [inline test/testoutput/3dvar_soca.test:22]: your mean values are outside the range of the min/max values. The norm is not getting calculated correctly somewhere

Comment (guillaumevernieres) [reply test/testoutput/3dvar_soca.test:22]: @travissluka : We are dividing by the number of unmasked grid point in gpnorm mean. Should be addressed in different a pr.

Comment (travissluka) [reply test/testoutput/3dvar_soca.test:22]: ah, ok. So then that should go away when we are pulling actual surface fields from MOM

**Review by travissluka — APPROVED** [review]

**Pull request merged**